src/main/java/jp/co/future/uroborosql/SqlAgentImpl.java

```diff
@@ -134,25 +137,29 @@ public ResultSet query(final SqlContext sqlContext) throws SQLException {
 				retryWaitTime = sqlContext.getRetryWaitTime();
 			}
 			int loopCount = 0;
+			Dialect dialect = getSqlConfig().getDialect();
 			ResultSet rs = null;
 			try {
 				do {
 					try {
-						if (maxRetryCount > 0 && getSqlConfig().getDialect().isRollbackToSavepointBeforeRetry()) {
+						if (maxRetryCount > 0 && dialect.isRollbackToSavepointBeforeRetry()) {
 							setSavepoint(RETRY_SAVEPOINT_NAME);
 						}
 						rs = new InnerResultSet(getSqlFilterManager().doQuery(sqlContext, stmt, stmt.executeQuery()),
 								stmt);
 						stmt.closeOnCompletion();
 						return rs;
 					} catch (SQLException ex) {
-						if (maxRetryCount > 0 && getSqlConfig().getDialect().isRollbackToSavepointBeforeRetry()) {
+						if (maxRetryCount > 0 && dialect.isRollbackToSavepointBeforeRetry()) {
 							rollback(RETRY_SAVEPOINT_NAME);
 						}
+						String errorCode = String.valueOf(ex.getErrorCode());
+						String sqlState = ex.getSQLState();
+						Set<String> pessimisticLockingErrorCodes = dialect.getPessimisticLockingErrorCodes();
 						if (maxRetryCount > loopCount) {
-							String errorCode = String.valueOf(ex.getErrorCode());
-							String sqlState = ex.getSQLState();
-							if (getSqlRetryCodes().contains(errorCode) || getSqlRetryCodes().contains(sqlState)) {
+							if (getSqlRetryCodes().contains(errorCode) || getSqlRetryCodes().contains(sqlState)
+									|| pessimisticLockingErrorCodes.contains(errorCode)
+									|| pessimisticLockingErrorCodes.contains(sqlState)) {
```

Comment:
@HidekiSugimoto189 
In the current implementation, in the case of pessimistic lock error codes, it is always retrying, but shouldn't it be retrying only when it is set as a retry target error code?

If only deadlock errors are specified in the retry error code, but pessimistic lock errors are not included, I think it would be better for users to have a specification where PessimisticLockException is thrown without retrying.

Reply:
@shout-star 
Thank you for the review.
Revised issues.

Reply:
@HidekiSugimoto189 
![エンジニアを褒めるネコ:LGTM](https://raw.githubusercontent.com/fukayatsu/lttm-crx/master/vendor/%E3%82%A8%E3%83%B3%E3%82%B8%E3%83%8B%E3%82%A2%E3%82%92%E8%A4%92%E3%82%81%E3%82%8B%E3%83%8D%E3%82%B3/01_LGTM.png)